### PR TITLE
ACR beta3 changes.

### PR DESCRIPTION
**Pull request by pallavit**

### CODE REVIEW

File: sdk/containerregistry/azure-containers-containerregistry/README.md [hunk 1]
```diff
@@ -102,19 +125,169 @@ ContainerRegistryAsyncClient client = new ContainerRegistryClientBuilder()
...
+
+    imageManifests.skip(imagesCountToKeep).subscribe(imageManifest -> {
+        System.out.printf(String.format("Deleting image with digest %s.%n", imageManifest.getDigest()));
+        System.out.printf("    This image has the following tags: ");
+
+        for (String tagName : imageManifest.getTags()) {
+            System.out.printf("        %s:%s", imageManifest.getRepositoryName(), tagName);
+        }
+
+        repository.getArtifact(imageManifest.getDigest()).delete().subscribe();
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/README.md:189]: Generally, having a `subscribe()` within another `subscribe()` is not a recommended pattern. Instead use `flatMap()`.

```java
client.listRepositoryNames()
         .map(repositoryName -> client.getRepository(repositoryName))
         .flatMap(repository -> repository.listManifests(..).skip(3))
         .flatMap(manifest -> { // process manifest})
         .subscribe();   
```

File: sdk/containerregistry/azure-containers-containerregistry/README.md [hunk 2]
```diff
@@ -26,38 +26,57 @@ Use the client library for Azure Container Registry to:
...
+More information at [Azure Container Registry portal][container_registry_create_portal]
 
-<!-- embedme ./src/samples/java/com/azure/containers/containerregistry/ReadmeSamples.java#L21-L25 -->
+<!-- embedme ./src/samples/java/com/azure/containers/containerregistry/ReadmeSamples.java#L29-L33 -->
+```Java
+    ContainerRegistryClient client = new ContainerRegistryClientBuilder()
+        .endpoint(endpoint)
+        .credential(credential)
+        .buildClient();
+}
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/README.md:46]: nit: this line should be removed.

File: sdk/containerregistry/azure-containers-containerregistry/README.md [hunk 3]
```diff
@@ -89,32 +116,110 @@ ContainerRegistryClient client = new ContainerRegistryClientBuilder()
...
+final String repositoryName = getRepositoryName();
+
+ContainerRegistryClient anonymousClient = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .buildClient();
+
+try {
+    anonymousClient.deleteRepository(repositoryName);
+    System.out.println("Unexpected Success: Delete is not allowed on anonymous access");
+} catch (Exception ex) {
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/README.md:205]: Use `ClientAuthenticationException` here to show that it throws a specific exception that is diagnosable.

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java [hunk 4]
```diff
@@ -162,36 +147,42 @@ public String getEndpoint() {
      * @param repositoryName Name of the image (including the namespace).
      * @return deleted repository properties.
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryAsyncClient.java:148]: Update javadoc

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryClient.java [hunk 5]
```diff
@@ -73,49 +68,54 @@ public String getEndpoint() {
...
-     * @param repositoryName Name of the repository (including the namespace).
-     * @param tagOrDigest Tag or digest associated with the artifact.
-     * @return repository client.
+     * @param repositoryName Name of the repository to reference.
+     * @param digest Either a tag or digest that uniquely identifies the artifact.
+     * @return A new {@link RegistryArtifact} object for the desired repository.
+     * @throws NullPointerException if 'repositoryName' is null.
+     * @throws IllegalArgumentException if 'repositoryName' is empty.
+     * @throws NullPointerException if 'digest' is null.
+     * @throws IllegalArgumentException if 'digest' is empty.
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryClient.java:116]: Don't need to repeat the exceptions here. Define the exception once and describe all cases which can throw that exception.

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifactAsync.java [hunk 6]
```diff
@@ -107,19 +134,14 @@ public String getRepositoryName() {
      * Gets the tag or digest for the current instance.
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/RegistryArtifactAsync.java:134]: Update javadoc

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/models/PathsV3R3RxOauth2TokenPostRequestbodyContentApplicationXWwwFormUrlencodedSchema.java [hunk 7]
```diff
@@ -10,12 +10,6 @@
 /** The PathsV3R3RxOauth2TokenPostRequestbodyContentApplicationXWwwFormUrlencodedSchema model. */
 @Fluent
 public final class PathsV3R3RxOauth2TokenPostRequestbodyContentApplicationXWwwFormUrlencodedSchema {
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/models/PathsV3R3RxOauth2TokenPostRequestbodyContentApplicationXWwwFormUrlencodedSchema.java:12]: :sweat_smile:

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/models/ArtifactManifestProperties.java [hunk 8]
```diff
@@ -57,63 +68,159 @@
...
+    /*
+     * Read enabled
+     */
+    @JsonProperty(value = "manifest.changeableAttributes.readEnabled")
+    private Boolean readEnabled;
+
+    /**
+     * Initializes an instance of {@link ArtifactManifestProperties}.
+     */
+    public ArtifactManifestProperties() { }
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/models/ArtifactManifestProperties.java:213]: This is not required.

File: sdk/containerregistry/azure-containers-containerregistry/src/samples/java/com/azure/containers/containerregistry/AnonymousAsyncClientThrows.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.containers.containerregistry;
+
+public class AnonymousAsyncClientThrows {
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/src/samples/java/com/azure/containers/containerregistry/AnonymousAsyncClientThrows.java:6]: Add javadoc

File: sdk/containerregistry/azure-containers-containerregistry/src/samples/java/com/azure/containers/containerregistry/AnonymousAsyncClientThrows.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+    public static void main(String[] args) {
+        ContainerRegistryAsyncClient anonymousClient = new ContainerRegistryClientBuilder()
+            .endpoint(ENDPOINT)
+            .buildAsyncClient();
+
+        anonymousClient.deleteRepository(REPOSITORY_NAME).subscribe(deleteRepositoryResult -> {
+            System.out.println("Unexpected Success: Delete is not allowed on anonymous access");
+        }, error -> {
+            System.out.println("Expected exception: Delete is not allowed on anonymous access");
```
Comment (srnagar) [inline sdk/containerregistry/azure-containers-containerregistry/src/samples/java/com/azure/containers/containerregistry/AnonymousAsyncClientThrows.java:19]: This should be printed out only if the error is of type "ClientAuthenticationException". If it is a different exception (like npe/illegal arg/timeout/connection errors), it is not expected.

**Review by srnagar — APPROVED** [review]
Looks good. Just update documentation in a few places and ship it!

File: sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/models/PathsV3R3RxOauth2TokenPostRequestbodyContentApplicationXWwwFormUrlencodedSchema.java (hunk 7 above)
Comment (pallavit) [reply sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/models/PathsV3R3RxOauth2TokenPostRequestbodyContentApplicationXWwwFormUrlencodedSchema.java:12]: Lol!

**Pull request merged**